### PR TITLE
Escape newlines in `StringLiteralExprSyntax(contents:)` initializer.

### PR DESCRIPTION
**Pull request by DougGregor**

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please test

**Review by bnbarham — APPROVED** [review]
Could add a `StringLiteralTests` test:
```
  func testNewlines() {
    AssertBuildResult(
      StringLiteralExpr(content: "linux\nwindows\r\n"), "linux\\nwindows\\r\\n"
    )
  }
```

LGTM otherwise though

EDIT: Actually, what about `\u`? Both in the general case and because there's a bunch of other newline characters

**Comment by DougGregor** [comment]
Great suggestions, @bnbarham , I've updated the PR accordingly.

**Comment by DougGregor** [comment]
@swift-ci please test

File: Sources/SwiftSyntaxBuilder/ConvenienceInitializers.swift [hunk 1]
```diff
@@ -192,6 +192,28 @@ extension MemberAccessExpr {
...
+    var result = ""
+    var input = self[...]
+    while let firstNewline = input.firstIndex(where: { $0.isNewline }) {
+      result += input[..<firstNewline]
+      if input[firstNewline] == "\r" {
+        result += "\\r"
+      } else if input[firstNewline] == "\r\n" {
+        result += "\\r\\n"
+      } else {
+        result += "\\n"
```
Comment (bnbarham) [inline Sources/SwiftSyntaxBuilder/ConvenienceInitializers.swift:207]: Replacing is probably fair enough, though you could also do `result += input[firstNewLine].unicodeScalars.map({ $0.escaped(asASCII: true) }).joined())` 🤷

**Pull request merged**